### DISCUSSION TITLE
Trouble with sootup.java.bytecode jar

### DISCUSSION THREAD

**Discussion by SoniaZaldana** [soot-oss/SootUp#567] [Q&A] [answered]
Hi folks, 

I'm working on adding some soot up dependencies to a modularized project (using Java 9+). It seemed pretty straight forward to deal with the sootup jars as automatic modules, but I've been running into a few issues specifically with the sootup.java.bytecode jar. 

Note how upon inspecting this jar, it appears as though it contains a module descriptor which does not match the module. 

```
jar --file=sootup.java.bytecode.jar --describe-module 
org.objectweb.asm.util@9.3.0 jar:file:///home/szaldana/SootUp/sootup.java.bytecode/target/sootup.java.bytecode-1.0.0-SNAPSHOT.jar!/module-info.class open
exports org.objectweb.asm.util
requires java.base mandated
requires org.objectweb.asm transitive
requires org.objectweb.asm.tree transitive
requires org.objectweb.asm.tree.analysis transitive

```

My project is also using some asm dependencies, so things got a bit complicated. I've been working around this issue by adding the following to the pom and generating my own jar. 

```
<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-jar-plugin</artifactId>
				<configuration>
					<archive>
						<manifestEntries>
							<Automatic-Module-Name>sootup.java.bytecode</Automatic-Module-Name>
						</manifestEntries>
					</archive>
				</configuration>
			</plugin>
		</plugins>
	</build>
```

Is there something obvious I could be missing? For the record, I am able to work with the other jars for soot up without any problems, as they don't pick up any existing module descriptors. 

Lastly, I just wanted to note that the link for what each module includes in the [getting started](https://soot-oss.github.io/SootUp/installation/) docs is broken. Resolving issues with split packages was a bit more difficult without it :)

Looking forward to your response!

Cheers, 
Sonia

**Comment by kadirayk** [comment, marked as answer]
Hi, this issue might be solved with the release v1.1.0, which doesn't contain fat jars.

**Reply by SoniaZaldana** [reply]
Thanks! v1.1.0 works :)
